src/System.Device.Gpio/System/Device/Gpio/GpioController.Linux.cs

```diff
@@ -30,6 +31,17 @@ public GpioController(PinNumberingScheme numberingScheme)
         /// <returns>A driver that works with the board the program is executing on.</returns>
         private static GpioDriver GetBestDriverForBoard()
         {
+            // a first chance of finding a raspberry pi..... perhaps not on Raspbian note 
+            // that we want this to fail quietly if something goes wrong with the test
+            try
```

Comment:
I think we will want to take the PR changes in for 3.0, but we don't want to take this particular change here as it may be risky. I think that if somebody wants to force the use of the RaspberryPiDriver if it is not picked by default they have a way of doing it, so it is probably dangerous to change this logic now as we don't know how other boards will behave with this change.

Reply:
@joperezr 

Do you really want to do this ? The implication of doing this test is that, by default, Debian and Ubuntu on a Raspberry Pi will end up using the Unix driver as only Raspbian has the SOC identifier in /proc/cpuinfo.

This does feel a bit of a contradiction as the some of idea of using /dev/mem is that other distributions may not implement /dev/gpiomem and without this change you would have to force the 

I also think that the risk is low as the chances of a non-raspberry pi having the text in the "/proc/device-tree/model" node is unlikely and in case of any exception then we just continue with the existing code.

Given this do you still want me to remove this area of the PR ?

Reply:
The thing that worries me is that we have done some extensive board testing in the past but we don't yet have all of these different boards in the lab, so we don't have automation against many boards yet. In our initial test run, we would have chosen a specific driver (like the SysFS or Libgpiod) in one board but if we add this logic we would be changing that and we are getting very close to release so I'm afraid this would be a risky change. The other thing that makes me feel like we may not want to take this very specific part of the change in, is that there is a workaroud, that I agree is not ideal, but the consumer can force the use of RaspberryPiDriver if they wanted to. Most of our device bindings that use GpioControllers take in a IGpioController in their constructor, so they could  have code in their end to detect a RaspberryPi, and if they did then they could pass in a controller that forced this driver.

I'm not saying that we will never take this change, I'm just saying that it may be risky to take this at this stage of the release where we will only have about 2 more previews before release. I actually think that we should have used your check in the beginning instead of using /proc/cpuinfo, but given that we have done extensive testing with that logic, I would rather not change this now. 

This is my current thinking, but I'm totally up for discussing this more if you feel strongly about this.

Reply:
@joperezr 
I have no issues with anything from above .... I will remove the detection logic and then let you know. Perhaps I should raise an issue that could be marked for 3.1

Reply:
Sounds like a plan, I would actually mention in the issue that what we want is to replace the existing logic that uses Regex and /proc/cpuinfo and use yours instead.

Reply:
@joperezr 
I have removed the Pi Detection logic... I have also tested that it still works no a Pi 4B on Raspbian by renaming /dev/gpiomem and seeing it accessing /dev/mem with the debugger. Hope that is o.k.